### PR TITLE
2fix panel fixed

### PR DESCRIPTION
**Pull request by chrafuse**

### CODE REVIEW

File: browser/components/loop/content/css/panel.css [hunk 1]
```diff
@@ -254,29 +278,23 @@ html[dir="rtl"] .tab-view li:nth-child(2).selected ~ .slide-bar {
...
-}
 
-.room-list-empty {
-  margin: 5% 15px;
+.no-conversations-message {
+  position: relative;
+  top: 50%;
+  transform: translateY(-50%);
+  padding-top: 75px;
+  padding-bottom: 10px;
```
Comment (dmose) [inline browser/components/loop/content/css/panel.css:296]: Please add a comment here, describing what's happening, perhaps even linking to the stackoverflow reference.

File: browser/components/loop/content/css/panel.css [hunk 2]
```diff
@@ -361,7 +379,8 @@ html[dir="rtl"] .tab-view li:nth-child(2).selected ~ .slide-bar {
   height: 3rem;
   display: block;
   font-size: 1.2rem;
-  margin: 0 15px 1rem 15px;
+  /*margin: 0 15px 1rem 15px;*/
```
Comment (dmose) [inline browser/components/loop/content/css/panel.css]: This can be deleted.

File: browser/components/loop/content/css/panel.css (hunk 1 above)
Comment (dmose) [inline browser/components/loop/content/css/panel.css:296]: Also maybe add a comment that the no-cvonerstions message container is only necessary for this centering stuff.  Perhaps even in the JSX?

File: browser/components/loop/content/css/panel.css [hunk 3]
```diff
@@ -394,9 +414,12 @@ html[dir="rtl"] .tab-view li:nth-child(2).selected ~ .slide-bar {
   font-size: 1.3rem;
   line-height: 2.4rem;
   color: #000;
+  /* See .room-entry-context-item for the margin/size reductions. */
+  /*width: calc(100% - 1rem - 16px);*/
```
Comment (dmose) [inline browser/components/loop/content/css/panel.css]: Above width can be deleted.

File: browser/components/loop/content/css/panel.css [hunk 4]
```diff
@@ -394,9 +414,12 @@ html[dir="rtl"] .tab-view li:nth-child(2).selected ~ .slide-bar {
...
   /* See .room-entry-context-item for the margin/size reductions.
-   * An extra 40px to make space for the call button and chevron. */
+    * An extra 40px to make space for the call button and chevron. */
```
Comment (dmose) [inline browser/components/loop/content/css/panel.css]: Indent off-by-one.

File: browser/components/loop/content/css/panel.css [hunk 5]
```diff
@@ -361,7 +379,8 @@ html[dir="rtl"] .tab-view li:nth-child(2).selected ~ .slide-bar {
   height: 3rem;
   display: block;
   font-size: 1.2rem;
-  margin: 0 15px 1rem 15px;
+  /*margin: 0 15px 1rem 15px;*/
+  margin: 7px 15px 0px;
```
Comment (dmose) [inline browser/components/loop/content/css/panel.css]: Local style is to never use any units on a 0.

File: browser/components/loop/content/css/panel.css [hunk 6]
```diff
@@ -493,27 +516,27 @@ html[dir="rtl"] .tab-view li:nth-child(2).selected ~ .slide-bar {
 
 /* Room entry context button (call button + chevron) */
 .room-entry-context-actions {
-  display: none;
-  border-radius: 30px;
-  background: #56b397;
-  vertical-align: top;
+   display: none;
```
Comment (dmose) [inline browser/components/loop/content/css/panel.css]: Too much indentation here & in many subsequent lines.

File: browser/components/loop/content/css/panel.css [hunk 7]
```diff
@@ -17,12 +17,27 @@ body {
...
+  align-content: stretch;
+  align-items: flex-start;
 }
 
 /* Notifications displayed over tabs */
 
 .panel .messages {
+  width: 100%;
   margin: 0;
+  /*display: none;*/
```
Comment (dmose) [inline browser/components/loop/content/css/panel.css]: This wants to be removed, I think.

File: browser/components/loop/content/css/contacts.css [hunk 8]
```diff
@@ -7,6 +7,67 @@ html {
...
+  flex-direction: column;
+  overflow: auto;
+}
+/* Don't show the Gravatar if we're showing a contacts list. */
+.contact-list ~ .contacts-gravatar-promo {
+  display: none;
+}
+
+.contact-list-wrapper {
+  flex: 2;
```
Comment (dmose) [inline browser/components/loop/content/css/contacts.css]: This may be able to be flex: 1

File: browser/components/loop/content/css/panel.css [hunk 9]
```diff
@@ -1047,4 +1067,4 @@ html[dir="rtl"] .settings-menu .dropdown-menu {
 .fte-get-started-button:active {
   background-color: #5cccee;
   color: #fff;
-}
+}
```
Comment (dmose) [inline browser/components/loop/content/css/panel.css]: Please put back the newline, if you would.

File: browser/components/loop/content/css/contacts.css [hunk 10]
```diff
@@ -7,6 +7,67 @@ html {
...
+  overflow-x: hidden;
+  overflow-y: auto;
+}
+
+/* Don't show the Gravatar if we're showing a contacts list. NOT WORKING hide in code*/
+.contact-search-list-empty ~ .contacts-gravatar-promo {
+  display: block;
+}
```
Comment (dmose) [inline browser/components/loop/content/css/contacts.css]: After a bit of playing around, it looks to me like this isn't needed at all.

Comment (dmose) [inline browser/components/loop/content/css/contacts.css]: I've just added a commit that fixes this.

File: browser/components/loop/content/css/contacts.css [hunk 11]
```diff
@@ -7,6 +7,67 @@ html {
...
+
+.contact-list-empty,
+.contact-search-list-empty {
+  margin-top: 4rem;
+  padding-top: 11.5rem;
+  padding-bottom: 0;
+  text-align: center;
+  color: #4a4a4a;
+}
```
Comment (dmose) [inline browser/components/loop/content/css/contacts.css]: This wants to be spun off into another bug cleanup, probably by using flex to the vertical centering.  Let's link to that bug in an XXX comment above this pair of selectors.

File: browser/components/loop/content/css/contacts.css [hunk 12]
```diff
@@ -7,6 +7,67 @@ html {
...
+
+.contact-search-list-empty {
+  background-image: url("../shared/img/empty_search.svg");
+}
+
+.contact-list-empty {
+  background-image: url("../shared/img/empty_contacts.svg");
+  flex: 1;
```
Comment (dmose) [inline browser/components/loop/content/css/contacts.css]: This flex seems to be unnecessary, since the parent does not have a display flex.

Comment (dmose) [inline browser/components/loop/content/css/contacts.css]: Fix commited.

File: browser/components/loop/content/css/contacts.css [hunk 13]
```diff
@@ -340,7 +378,10 @@ html[dir="rtl"] .contact > .dropdown-menu {
 }
 
 .contact-form {
-  padding: 14px; /* Override based on spacing in Mockup */
+  padding: 14px 15px 0; /* Override based on spacing in Mockup */
+  flex: 2 1 0%;
```
Comment (dmose) [inline browser/components/loop/content/css/contacts.css]: flex: 1 works here

File: browser/components/loop/content/css/contacts.css [hunk 14]
```diff
@@ -340,7 +378,10 @@ html[dir="rtl"] .contact > .dropdown-menu {
 }
 
 .contact-form {
-  padding: 14px; /* Override based on spacing in Mockup */
+  padding: 14px 15px 0; /* Override based on spacing in Mockup */
```
Comment (dmose) [inline browser/components/loop/content/css/contacts.css]: Let's use the four-value version here, the three-value version is so rarely seen around here that I think it's likely to be misread.

File: browser/components/loop/content/css/contacts.css [hunk 15]
```diff
@@ -352,9 +393,7 @@ html[dir="rtl"] .contact > .dropdown-menu {
 }
 
 .contact-form .form-content-container {
-  height: 24.1rem; /* This height is needed to keep the panel height at 400px and
-                      the bottom elements at the bottom of the panel
-                      Can likely go away if we switched this pane to flexbox model */
+  flex: 1;
```
Comment (dmose) [inline browser/components/loop/content/css/contacts.css]: Add a comment about this anchoring buttons to the area of the footer.

File: browser/components/loop/content/css/panel.css [hunk 16]
```diff
@@ -17,12 +17,27 @@ body {
...
+
+/* Panel container flexbox */
+.panel-content {
+  height: 410px;
+  width: 330px;
+  display: flex;
+  flex-flow: column nowrap;
+  justify-content: flex-start;
+  align-content: stretch;
```
Comment (dmose) [inline browser/components/loop/content/css/panel.css]: Above two lines are the defaults; can be deleted.  I've done so.

File: browser/components/loop/content/css/panel.css (hunk 7 above)
Comment (dmose) [inline browser/components/loop/content/css/panel.css]: Commited.

File: browser/components/loop/content/css/panel.css [hunk 17]
```diff
@@ -67,10 +82,16 @@ body {
...
 .tab-view-container {
+  width: 100%;
   background-image: url("../shared/img/beta-ribbon.svg#beta-ribbon");
   background-color: #fbfbfb;
   background-size: 36px 36px;
   background-repeat: no-repeat;
+  flex: 1;
+  display: flex;
+  flex-flow: column nowrap;
+  height: 368px;
```
Comment (dmose) [inline browser/components/loop/content/css/panel.css]: Leave an XXX comment suggest that we might be able to drop this rule.

File: browser/components/loop/content/css/panel.css [hunk 18]
```diff
@@ -179,7 +200,7 @@ html[dir="rtl"] .tab-view li:nth-child(2).selected ~ .slide-bar {
   background-image: url("../shared/img/icons-14x14.svg#hello-hover");
 }
 
-.tab-view li:first-child:nth-last-child(2) ~ div {
+.tab-view li:first-child:nth-last-child(2) ~ li {
```
Comment (dmose) [inline browser/components/loop/content/css/panel.css:199]: A comment explaining why/what would  be helpful.

File: browser/components/loop/content/css/contacts.css [hunk 19]
```diff
@@ -7,6 +7,67 @@ html {
...
+  overflow-y: auto;
+}
+
+.contact-list-empty,
+.contact-search-list-empty {
+  margin-top: 4rem;
+  padding-top: 11.5rem;
+  padding-bottom: 0;
+  text-align: center;
+  color: #4a4a4a;
```
Comment (dmose) [inline browser/components/loop/content/css/contacts.css]: Please remove these two lines, as they're already in the shared code.

File: browser/components/loop/content/css/panel.css [hunk 20]
```diff
@@ -370,7 +390,7 @@ html[dir="rtl"] .tab-view li:nth-child(2).selected ~ .slide-bar {
   height: 3rem;
   display: block;
   font-size: 1.2rem;
-  margin: 0 15px 1rem 15px;
+  margin: 7px 15px 0;
```
Comment (dmose) [inline browser/components/loop/content/css/panel.css]: For this and the padding rule above, please switch to 4-value syntax.

File: browser/components/loop/content/css/panel.css [hunk 21]
```diff
@@ -382,14 +402,18 @@ html[dir="rtl"] .tab-view li:nth-child(2).selected ~ .slide-bar {
 }
 
 .room-list {
-  max-height: 335px; /* XXX better computation needed */
-  /* At least high enough to be able to contain the conversation menu. */
-  min-height: 100px;
-  overflow: auto;
+  /* flex grow full definition necessary */
+  flex: 1 1 0;
```
Comment (dmose) [inline browser/components/loop/content/css/panel.css]: Please make the above comment a bit clearer.

File: browser/components/loop/content/css/panel.css [hunk 22]
```diff
@@ -382,14 +402,18 @@ html[dir="rtl"] .tab-view li:nth-child(2).selected ~ .slide-bar {
...
+  overflow-x: hidden;
+  display: flex;
+  flex-flow: column nowrap;
 }
 
-.room-list:empty {
+.room-list-empty {
   border-bottom-width: 0;
+  flex: 1;
+  transform-style: preserve-3d;
```
Comment (dmose) [inline browser/components/loop/content/css/panel.css]: Please add a comment here linking to the blog showing why we're doing this.

File: browser/components/loop/content/css/panel.css [hunk 23]
```diff
@@ -551,11 +578,11 @@ html[dir="rtl"] .room-entry-call-btn {
 
 html[dir="rtl"] .room-entry-context-actions > .dropdown-menu {
   right: auto;
-  left: 21px;
+  left: 45px;
 }
 
 .room-entry-context-actions > .dropdown-menu {
-  right: 21px;
+  right: 45px;
```
Comment (dmose) [inline browser/components/loop/content/css/panel.css:588]: Please make these two things part of the other spin-out bug: why is this 45px?  should the word context be changed or gotten rid of (it seems wrong).

File: browser/components/loop/content/css/panel.css [hunk 24]
```diff
@@ -980,9 +1007,10 @@ html[dir="rtl"] .settings-menu .dropdown-menu {
   align-content: stretch;
   align-items: center;
   font-size: 1rem;
-  background-color: #fbfbfb;
   color: #666666;
-  padding: .5rem 15px;
+  padding: 10px 15px 6px;
```
Comment (dmose) [inline browser/components/loop/content/css/panel.css]: 4 value form please.

File: browser/components/loop/ui/ui-showcase.jsx [hunk 25]
```diff
@@ -283,7 +283,8 @@
   var roomStore = new loop.store.RoomStore(dispatcher, {
     mozLoop: navigator.mozLoop,
     activeRoomStore: makeActiveRoomStore({
-      roomState: ROOM_STATES.HAS_PARTICIPANTS
+      roomState: ROOM_STATES.HAS_PARTICIPANTS,
+      pendingInitialRetrieval: false
```
Comment (dmose) [inline browser/components/loop/ui/ui-showcase.jsx]: Check this out; I think we need to use setStoreState or just remove.

**Comment by dmose** [comment]
r=dmose once the above issues are addressed.  Ideally, we'd spin-off hte bugs before landing, but if that bleeds into tomorrow, that's fine.

File: browser/components/loop/content/css/contacts.css (hunk 8 above)
Comment (chrafuse) [inline browser/components/loop/content/css/contacts.css]: works and done.

File: browser/components/loop/content/css/contacts.css (hunk 11 above)
Comment (chrafuse) [inline browser/components/loop/content/css/contacts.css]: Removed duplicate css from contacts and left panel.css shared css properties. Done

File: browser/components/loop/content/css/contacts.css (hunk 13 above)
Comment (chrafuse) [inline browser/components/loop/content/css/contacts.css]: changed to flex 1. done

File: browser/components/loop/content/css/contacts.css (hunk 14 above)
Comment (chrafuse) [inline browser/components/loop/content/css/contacts.css]: 4 value version done.

File: browser/components/loop/content/css/contacts.css (hunk 15 above)
Comment (chrafuse) [inline browser/components/loop/content/css/contacts.css]: Comment added. Done

File: browser/components/loop/content/css/panel.css (hunk 17 above)
Comment (chrafuse) [inline browser/components/loop/content/css/panel.css]: Removed the height and tested in panel.  Inspected chrome computed height and gave correct adjusted heights. Done.

File: browser/components/loop/content/css/panel.css (hunk 18 above)
Comment (chrafuse) [inline browser/components/loop/content/css/panel.css:199]: Comment added about hiding tab-slider if one tab is shown. Done.

File: browser/components/loop/content/css/contacts.css (hunk 19 above)
Comment (chrafuse) [inline browser/components/loop/content/css/contacts.css]: Done.

File: browser/components/loop/content/css/panel.css (hunk 20 above)
Comment (chrafuse) [inline browser/components/loop/content/css/panel.css]: Done.

File: browser/components/loop/content/css/panel.css (hunk 21 above)
Comment (chrafuse) [inline browser/components/loop/content/css/panel.css]: Comment expanded to explain why this setting is used. Done

File: browser/components/loop/content/css/panel.css (hunk 22 above)
Comment (chrafuse) [inline browser/components/loop/content/css/panel.css]: Added comment about vertical align solution. Done.

File: browser/components/loop/content/css/panel.css (hunk 23 above)
Comment (chrafuse) [inline browser/components/loop/content/css/panel.css:588]: Added to user story list as new spin off bug . Done.

File: browser/components/loop/content/css/panel.css (hunk 24 above)
Comment (chrafuse) [inline browser/components/loop/content/css/panel.css]: Done.

File: browser/components/loop/ui/ui-showcase.jsx (hunk 25 above)
Comment (chrafuse) [inline browser/components/loop/ui/ui-showcase.jsx]: Removed and Done.

File: browser/components/loop/content/css/panel.css (hunk 1 above)
Comment (chrafuse) [inline browser/components/loop/content/css/panel.css:296]: Done

**Comment by chrafuse** [comment]
Rebased from 
Cmach interrupted by signal or user action. Stopping.
 4:07.02 TEST_END: Thread-23 FAIL
 4:07.02 LOG: Thread-23 INFO browser/components/loop/test/xpcshell/test_loopservice_token_validation.js failed or timed out, will retry.

**Comment by dmose** [comment]
What does the most recent comment mean?

**Comment by chrafuse** [comment]
This should have been closed a while back.

**Pull request closed without merging**